### PR TITLE
[263] API-functional tests

### PR DESCRIPTION
**Pull request by VoronoyAlexandr**
### Description (*)
This PR add API-function tests which cover negative scenarios
### Fixed Issues (if relevant)
1. magento/graphql-ce#263: Test coverage for PR-226

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+mutation {
+    sendEmailToFriend(
+        input: {
+          product_id: 1
+          sender: {
+            name: "Name"
+            email: "e@mail.com"
+            message: "Lorem Ipsum"
+        }
+          recipients: [
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php:146]: Looks like two elements of recipients will be enough

File: dev/tests/integration/testsuite/Magento/SendFriend/_files/product_simple.php [hunk 2]
```diff
@@ -0,0 +1,212 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+use Magento\Catalog\Api\Data\ProductTierPriceExtensionFactory;
+use Magento\Catalog\Api\Data\ProductExtensionInterfaceFactory;
+
+\Magento\TestFramework\Helper\Bootstrap::getInstance()->reinitialize();
```
Comment (naydav) [inline dev/tests/integration/testsuite/Magento/SendFriend/_files/product_simple.php:10]: Why do we need to have this fixture if we already have same fixture in Magento code base?

File: dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+            name
+            email
+        }
+    }
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage("You can't send messages more than {$sendFriend->getMaxSendsToFriend()} times an hour.");
+
+        for ($i = 0; $i <= $sendFriend->getMaxSendsToFriend() + 1; $i++) {
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php:465]: So, if value will be `10000` we will have huge cycle in tests

File: dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+    /**
+     * @var SendFriendFactory
+     */
+    private $sendFriendFactory;
+    /**
+     * @var DataObjectFactory
+     */
+    private $dataObjectFactory;
+
+    protected function setUp()
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php:28]: Pls, remove unused properties

File: dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage("No more than {$sendFriend->getMaxRecipients()} emails can be sent at a time.");
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/SendFriend/_files/product_simple.php
+     */
+    public function testSendWithoutRecipentsName()
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php:194]: Looks like for validation w can use `dataProviders` instead of creating separate methods

File: dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Please provide Message.');
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/SendFriend/_files/product_simple.php
+     * @magentoApiDataFixture Magento/SendFriend/Fixtures/sendfriend_configuration.php
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php:418]: Generally, we should use the approach described in `https://github.com/magento/graphql-ce/issues/167`
Pls, link this temporal solution with the https://github.com/magento/graphql-ce/issues/167

**Review by naydav — CHANGES_REQUESTED** [review]

**Comment by naydav** [comment]
Hello @VoronoyAlexandr,
We still need some changes to finish the PR, will you continue progress? 
Thank you

Please let me know if you need any assistance.

**Comment by VitaliyBoyko** [comment]
Hi
Closing due inactivity.

File: dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php (hunk 1 above)
Comment (VitaliyBoyko) [reply dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php:146]: No
It's ok
Check app/code/Magento/SendFriend/etc/config.xml:15

File: dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php (hunk 3 above)
Comment (VitaliyBoyko) [reply dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php:465]: It is not clear so far how this ```10000``` can appear :).

File: dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php (hunk 6 above)
Comment (VitaliyBoyko) [reply dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php:418]: As far as I understand I can skip this comment. Just leave todo.
